src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -769,6 +758,22 @@ CHIP_ERROR ThermostatAttrAccess::Read(const ConcreteReadAttributePath & aPath, A
         Delegate * delegate = GetDelegate(aPath.mEndpointId);
         VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Delegate is null"));
 
+        if (InAtomicWrite(aPath.mEndpointId))
```

Comment:
Doesn't this need to also check that `GetAtomicWriteScopedNodeId(aPath.mEndpointId)` matched the reader's scoped node id?    Something like:
```suggestion
        auto & desc = aEncoder.GetSubjectDescriptor();
        if (InAtomicWrite(aPath.mEndpointId) && desc.authMode == Access::AuthMode::kCase && GetAtomicWriteScopedNodeId(aPath.mEndpointId) == ScopedNodeId(desc.subject, desc.fabricIndex))
```